andela/annie-hall-cfh
# #152798651 Add emojis to in-game chat window

### What does this PR do?

It adds emojis as available chat options to in-game chat window

### Description of Task to be completed?

* Download and integrate `emojionearea` librabry in application
* Add jquery function to initialize the library and set event functionality when an emoji is choosen

### How should this be manually tested?

```
 1. From your terminal or console window, run brew install yarn
 2. Complete the basic setup of the project gotten from the README, on your local repository.
 3. Cd to the root of the project folder in your console.
 4. First run the command "yarn install" to install all required and updated node packages including
    bower components
 5. Log on to firebase and create a database. Make sure read and write permissions to the database 
     are enabled
 6. Create a .env file in root of project folder and add 
      MONGOHQ_URL=mongodb://anniehall:anniehall123@ds113826.mlab.com:13826/cfh_dev
      JWT_SECRET=anyRandomWordsOfYourChoose     
      RANDOM_CHARACTERS=anyRandomCharactersOfYourChoice
      API_KEY=yourFirebaseApiKey
      AUTH_DOMAIN=yourFirebaseAuthDomain
      DATABASE_URL=yourfirebaseDatabaseUrl
      PROJECT_ID=yourFirebaseProjectID
      STORAGE_BUCKET=yourFirebaseStorageBucket
      MESSAGING_SENDER_ID=yourFirebaseMessagingSenderID
    inside the file.
 7. Run the command `yarn global add gulp-cli`
 8. Run the command `yarn start` or `gulp build && gulp` to build and start the application.
 9. Navigate to your browser and enter localhost:3000
 10. Invite other players (atleast 3 or atmost 11) to your game session.
 11. Click the chat icon container
 12. Send message by clicking enter after typing in a chat phrase or letter or(and) selecting an emoji 
     from the dropdown

```

Any background context you want to provide?

Nil

### What are the relevant pivotal tracker stories?

story type: feature
story Id: 152798651
story title: Users should be able to use emojis in the chat window

### Screenshots (if appropriate)

<img width="1440" alt="screen shot 2017-12-14 at 8 53 50 pm" src="https://user-images.githubusercontent.com/22524619/34012065-a155c2fa-e112-11e7-8e51-2c476044e1ea.png">

<img width="1440" alt="screen shot 2017-12-14 at 8 54 37 pm" src="https://user-images.githubusercontent.com/22524619/34012074-aae78f38-e112-11e7-90d5-b99c1b9e6a4f.png">

<img width="1440" alt="screen shot 2017-12-14 at 8 55 15 pm" src="https://user-images.githubusercontent.com/22524619/34012088-b894b2e6-e112-11e7-9f7d-15c295139321.png">
